course/models.py

```diff
@@ -722,17 +723,28 @@ def clean(self):
             errors['opening_time'] = _("Opening time must be earlier than the closing time.")
         if self.late_submissions_allowed and self.late_submission_deadline <= self.closing_time:
             errors['late_submission_deadline'] = _("Late submission deadline must be later than the closing time.")
+        if self.reading_opening_time and self.reading_opening_time > self.opening_time:
+            errors['reading_opening_time'] = _("Opening time of reading material "
+                "must be earlier than the opening time of the exercises.")
         if errors:
             raise ValidationError(errors)
 
     def is_open(self, when=None):
         when = when or timezone.now()
+        if self.reading_opening_time:
```

Comment:
Are there any consequences that the module is considered to be open already during the reading time? Now, exercises may not be submitted even though this method says that the module is open.

Reply:
I see you added a separate method for checking the exercise opening time. I just wonder if old code uses the right method for each use case then.